leshan-core/src/main/java/org/eclipse/leshan/core/node/codec/tlv/LwM2mNodeTlvEncoder.java

```diff
@@ -185,7 +186,7 @@ private Tlv encodeResource(LwM2mResource resource) {
             case OBJLNK:
                 return TlvEncoder.encodeObjlnk((ObjectLink) value);
```

Comment:
A try/catch block (for IllegalArgumentException) should be added here around all TlvEncoder.encode() methods. 
And it may be cleaner to raise TlvException instead of IllegalArgumentException in the TlvEncoder class.